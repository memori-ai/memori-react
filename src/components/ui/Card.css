.memori-card {
    display: flex;
    overflow: hidden;
    flex-direction: column;
    border-radius: 10px;
    margin: 0 0 1rem;
    background-color: #fff;
    box-shadow: 0 0.5rem 1rem 0 rgba(0, 0, 0, 0.1);
}

.memori-card--hoverable {
    transition: box-shadow 0.3s;
}

.memori-card--hoverable:hover {
    box-shadow: 0 0.2rem 0 rgba(0, 0, 0, 0.1), 0 0.5rem 1rem 0 rgba(0, 0, 0, 0.2);
}

.memori-card--hoverable:focus,
.memori-card--hoverable:focus-within {
    outline-color: var(--memori-primary);
}

.memori-card--cover {
    display: flex;
    overflow: hidden;
    height: 200px;
    align-items: center;
    justify-content: center;
    border-radius: 10px 10px 0 0;
    background-color: #eee;
}

.memori-card--cover img {
    min-width: 100%;
    min-height: 100%;
    object-fit: cover;
    object-position: center;
}

.memori-card--content {
    display: flex;
    flex: 1;
    flex-direction: column;
    justify-content: space-between;
    padding: 1.5rem 1rem;
}

.memori-card--title {
    margin: 0 0 0.5rem;
    font-size: 1.25rem;
    font-weight: 600;
}

.memori-card--description {
    margin: 0;
    color: #999;
    font-size: 0.875rem;
    font-weight: 400;
}
