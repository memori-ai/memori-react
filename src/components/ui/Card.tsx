import React, { FC } from 'react';
import cx from 'classnames';
import Spin from './Spin';

export interface Props {
  children?: React.ReactChild;
  loading?: boolean;
  className?: string;
  title?: string;
  description?: string;
  cover?: JSX.Element | React.ReactNode | string;
  hoverable?: boolean;
}

const Card: FC<Props> = ({
  loading = false,
  className,
  title,
  description,
  cover,
  hoverable = false,
  children,
}) => (
  <div
    className={cx('memori-card', className, {
      'memori-card--loading': loading,
      'memori-card--with-cover': cover,
      'memori-card--hoverable': hoverable,
    })}
  >
    <Spin spinning={loading}>
      {cover && <div className="memori-card--cover">{cover}</div>}
      <div className="memori-card--content">
        {title && <h3 className="memori-card--title">{title}</h3>}
        {description && (
          <p className="memori-card--description">{description}</p>
        )}
        <div className="memori-card--children">{children}</div>
      </div>
    </Spin>
  </div>
);

export default Card;
