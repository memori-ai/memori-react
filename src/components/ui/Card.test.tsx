import React from 'react';
import { render } from '@testing-library/react';
import Card from './Card';

it('renders Card unchanged', () => {
  const { container } = render(<Card />);
  expect(container).toMatchSnapshot();
});

it('renders Card with title unchanged', () => {
  const { container } = render(<Card title="Card title" />);
  expect(container).toMatchSnapshot();
});

it('renders Card with description unchanged', () => {
  const { container } = render(<Card description="Card description" />);
  expect(container).toMatchSnapshot();
});

it('renders Card with cover unchanged', () => {
  const { container } = render(<Card cover={<img src="#" alt="" />} />);
  expect(container).toMatchSnapshot();
});

it('renders Card with children unchanged', () => {
  const { container } = render(
    <Card>
      <p>Ciao</p>
    </Card>
  );
  expect(container).toMatchSnapshot();
});

it('renders Card loading unchanged', () => {
  const { container } = render(<Card title="Card title" loading />);
  expect(container).toMatchSnapshot();
});

it('renders Card with custom CSS class unchanged', () => {
  const { container } = render(<Card className="lorem-ipsum" />);
  expect(container).toMatchSnapshot();
});

it('renders Card hoverable unchanged', () => {
  const { container } = render(<Card title="lorem-ipsum" hoverable />);
  expect(container).toMatchSnapshot();
});
