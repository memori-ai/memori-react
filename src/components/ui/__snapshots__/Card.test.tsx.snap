// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders Card hoverable unchanged 1`] = `
<div>
  <div
    class="memori-card memori-card--hoverable"
  >
    <div
      class="memori-spin"
    >
      <div
        class="memori-card--content"
      >
        <h3
          class="memori-card--title"
        >
          lorem-ipsum
        </h3>
        <div
          class="memori-card--children"
        />
      </div>
      <div
        class="memori-spin--spinner"
      >
        <svg
          aria-hidden="true"
          class="memori-loading-icon"
          focusable="false"
          role="img"
          viewBox="0 0 1024 1024"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M988 548c-19.9 0-36-16.1-36-36 0-59.4-11.6-117-34.6-171.3a440.45 440.45 0 00-94.3-139.9 437.71 437.71 0 00-139.9-94.3C629 83.6 571.4 72 512 72c-19.9 0-36-16.1-36-36s16.1-36 36-36c69.1 0 136.2 13.5 199.3 40.3C772.3 66 827 103 874 150c47 47 83.9 101.8 109.7 162.7 26.7 63.1 40.2 130.2 40.2 199.3.1 19.9-16 36-35.9 36z"
          />
        </svg>
      </div>
    </div>
  </div>
</div>
`;

exports[`renders Card loading unchanged 1`] = `
<div>
  <div
    class="memori-card memori-card--loading"
  >
    <div
      class="memori-spin memori-spin--spinning"
    >
      <div
        class="memori-card--content"
      >
        <h3
          class="memori-card--title"
        >
          Card title
        </h3>
        <div
          class="memori-card--children"
        />
      </div>
      <div
        class="memori-spin--spinner"
      >
        <svg
          aria-hidden="true"
          class="memori-loading-icon"
          focusable="false"
          role="img"
          viewBox="0 0 1024 1024"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M988 548c-19.9 0-36-16.1-36-36 0-59.4-11.6-117-34.6-171.3a440.45 440.45 0 00-94.3-139.9 437.71 437.71 0 00-139.9-94.3C629 83.6 571.4 72 512 72c-19.9 0-36-16.1-36-36s16.1-36 36-36c69.1 0 136.2 13.5 199.3 40.3C772.3 66 827 103 874 150c47 47 83.9 101.8 109.7 162.7 26.7 63.1 40.2 130.2 40.2 199.3.1 19.9-16 36-35.9 36z"
          />
        </svg>
      </div>
    </div>
  </div>
</div>
`;

exports[`renders Card unchanged 1`] = `
<div>
  <div
    class="memori-card"
  >
    <div
      class="memori-spin"
    >
      <div
        class="memori-card--content"
      >
        <div
          class="memori-card--children"
        />
      </div>
      <div
        class="memori-spin--spinner"
      >
        <svg
          aria-hidden="true"
          class="memori-loading-icon"
          focusable="false"
          role="img"
          viewBox="0 0 1024 1024"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M988 548c-19.9 0-36-16.1-36-36 0-59.4-11.6-117-34.6-171.3a440.45 440.45 0 00-94.3-139.9 437.71 437.71 0 00-139.9-94.3C629 83.6 571.4 72 512 72c-19.9 0-36-16.1-36-36s16.1-36 36-36c69.1 0 136.2 13.5 199.3 40.3C772.3 66 827 103 874 150c47 47 83.9 101.8 109.7 162.7 26.7 63.1 40.2 130.2 40.2 199.3.1 19.9-16 36-35.9 36z"
          />
        </svg>
      </div>
    </div>
  </div>
</div>
`;

exports[`renders Card with children unchanged 1`] = `
<div>
  <div
    class="memori-card"
  >
    <div
      class="memori-spin"
    >
      <div
        class="memori-card--content"
      >
        <div
          class="memori-card--children"
        >
          <p>
            Ciao
          </p>
        </div>
      </div>
      <div
        class="memori-spin--spinner"
      >
        <svg
          aria-hidden="true"
          class="memori-loading-icon"
          focusable="false"
          role="img"
          viewBox="0 0 1024 1024"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M988 548c-19.9 0-36-16.1-36-36 0-59.4-11.6-117-34.6-171.3a440.45 440.45 0 00-94.3-139.9 437.71 437.71 0 00-139.9-94.3C629 83.6 571.4 72 512 72c-19.9 0-36-16.1-36-36s16.1-36 36-36c69.1 0 136.2 13.5 199.3 40.3C772.3 66 827 103 874 150c47 47 83.9 101.8 109.7 162.7 26.7 63.1 40.2 130.2 40.2 199.3.1 19.9-16 36-35.9 36z"
          />
        </svg>
      </div>
    </div>
  </div>
</div>
`;

exports[`renders Card with cover unchanged 1`] = `
<div>
  <div
    class="memori-card memori-card--with-cover"
  >
    <div
      class="memori-spin"
    >
      <div
        class="memori-card--cover"
      >
        <img
          alt=""
          src="#"
        />
      </div>
      <div
        class="memori-card--content"
      >
        <div
          class="memori-card--children"
        />
      </div>
      <div
        class="memori-spin--spinner"
      >
        <svg
          aria-hidden="true"
          class="memori-loading-icon"
          focusable="false"
          role="img"
          viewBox="0 0 1024 1024"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M988 548c-19.9 0-36-16.1-36-36 0-59.4-11.6-117-34.6-171.3a440.45 440.45 0 00-94.3-139.9 437.71 437.71 0 00-139.9-94.3C629 83.6 571.4 72 512 72c-19.9 0-36-16.1-36-36s16.1-36 36-36c69.1 0 136.2 13.5 199.3 40.3C772.3 66 827 103 874 150c47 47 83.9 101.8 109.7 162.7 26.7 63.1 40.2 130.2 40.2 199.3.1 19.9-16 36-35.9 36z"
          />
        </svg>
      </div>
    </div>
  </div>
</div>
`;

exports[`renders Card with custom CSS class unchanged 1`] = `
<div>
  <div
    class="memori-card lorem-ipsum"
  >
    <div
      class="memori-spin"
    >
      <div
        class="memori-card--content"
      >
        <div
          class="memori-card--children"
        />
      </div>
      <div
        class="memori-spin--spinner"
      >
        <svg
          aria-hidden="true"
          class="memori-loading-icon"
          focusable="false"
          role="img"
          viewBox="0 0 1024 1024"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M988 548c-19.9 0-36-16.1-36-36 0-59.4-11.6-117-34.6-171.3a440.45 440.45 0 00-94.3-139.9 437.71 437.71 0 00-139.9-94.3C629 83.6 571.4 72 512 72c-19.9 0-36-16.1-36-36s16.1-36 36-36c69.1 0 136.2 13.5 199.3 40.3C772.3 66 827 103 874 150c47 47 83.9 101.8 109.7 162.7 26.7 63.1 40.2 130.2 40.2 199.3.1 19.9-16 36-35.9 36z"
          />
        </svg>
      </div>
    </div>
  </div>
</div>
`;

exports[`renders Card with description unchanged 1`] = `
<div>
  <div
    class="memori-card"
  >
    <div
      class="memori-spin"
    >
      <div
        class="memori-card--content"
      >
        <p
          class="memori-card--description"
        >
          Card description
        </p>
        <div
          class="memori-card--children"
        />
      </div>
      <div
        class="memori-spin--spinner"
      >
        <svg
          aria-hidden="true"
          class="memori-loading-icon"
          focusable="false"
          role="img"
          viewBox="0 0 1024 1024"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M988 548c-19.9 0-36-16.1-36-36 0-59.4-11.6-117-34.6-171.3a440.45 440.45 0 00-94.3-139.9 437.71 437.71 0 00-139.9-94.3C629 83.6 571.4 72 512 72c-19.9 0-36-16.1-36-36s16.1-36 36-36c69.1 0 136.2 13.5 199.3 40.3C772.3 66 827 103 874 150c47 47 83.9 101.8 109.7 162.7 26.7 63.1 40.2 130.2 40.2 199.3.1 19.9-16 36-35.9 36z"
          />
        </svg>
      </div>
    </div>
  </div>
</div>
`;

exports[`renders Card with title unchanged 1`] = `
<div>
  <div
    class="memori-card"
  >
    <div
      class="memori-spin"
    >
      <div
        class="memori-card--content"
      >
        <h3
          class="memori-card--title"
        >
          Card title
        </h3>
        <div
          class="memori-card--children"
        />
      </div>
      <div
        class="memori-spin--spinner"
      >
        <svg
          aria-hidden="true"
          class="memori-loading-icon"
          focusable="false"
          role="img"
          viewBox="0 0 1024 1024"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M988 548c-19.9 0-36-16.1-36-36 0-59.4-11.6-117-34.6-171.3a440.45 440.45 0 00-94.3-139.9 437.71 437.71 0 00-139.9-94.3C629 83.6 571.4 72 512 72c-19.9 0-36-16.1-36-36s16.1-36 36-36c69.1 0 136.2 13.5 199.3 40.3C772.3 66 827 103 874 150c47 47 83.9 101.8 109.7 162.7 26.7 63.1 40.2 130.2 40.2 199.3.1 19.9-16 36-35.9 36z"
          />
        </svg>
      </div>
    </div>
  </div>
</div>
`;
